server/routes/public.js

```diff
@@ -29,15 +29,48 @@ router.get('/api/login', async (req, res) => {
 });
 
 router.get('/api/create-user', async (req, res) => {
+    // joi verification
+    // Validate appropriate parameters are passed into the create account endpoint.
+    const paramSchema = Joi.object({
+       
+       //The following joi verifications make sure the values inputted adhere to the format requirements
+       //specified in the migrations> create-worker-table and migrations> create-student-table.
+
+        email: Joi.string().email({ minDomainSegments: 2, tlds: { allow: ['com', 'net', 'ca'] } }).max(320).required(),
+        password: Joi.string().min(3).max(40).required(),
+        phone: Joi.string().max(20).phoneNumber(),
+        firstName: Joi.string().min(1).max(20).required(),
+        lastName: Joi.string().min(1).max(20).required(),
+        studentID: Joi.number().positive().integer(),
+        workerID: Joi.number().positive().integer(),
+        type: Joi.string().valid(['student', 'worker']),
+        specialization: Joi.string().valid(['social worker', 'guidance councellor']),//verify options
+        accessToken: Joi.string().alphanum().min(3).max(30).required(), 
+        //insert joi verification for firstname,lastname, type,studentID,workerID,phone
```

Comment:
Hi Nancy,
It might be beneficial to add joi validation to ensure that the values are disjoint. For instance, if studentId is specified, then workerId should not be specified but ONE of them must be specified. Refer to the `appointments` endpoint in the private file for an example.

Reply:
Hi Amy,
Thank you for suggesting that!
My action: I've ensured that there is at least one key in the object by using object().min(1). `userType: Joi.string().min(1).valid(['student', 'worker']).
`
You can now see these changes inline 46 of server/routes/public.js

Please confirm if this takes care of your suggestion!

Reply:
Hi Nancy,
Makes sense. Thank you for updating the validation!